tensorflow_lite_support/ios/test/task/vision/object_detector/TFLObjectDetectorTests.m

```diff
@@ -0,0 +1,130 @@
+       expectedFirstLabel:@"cat"];
+  [self verfiyDetection:detectionResult.detections[1]
+      expectedBoundingBox:CGRectMake(602, 157, 394, 447)
+       expectedFirstScore:0.609375
+       expectedFirstLabel:@"cat"];
+  [self verfiyDetection:detectionResult.detections[2]
+      expectedBoundingBox:CGRectMake(260, 394, 179, 209)
+       expectedFirstScore:0.5625
+       expectedFirstLabel:@"cat"];
+  [self verfiyDetection:detectionResult.detections[3]
+      expectedBoundingBox:CGRectMake(387, 197, 281, 409)
+       expectedFirstScore:0.488281
+       expectedFirstLabel:@"dog"];
+}
+
+- (void)verifyResultsWithMaxResultsOption:(TFLDetectionResult *)detectionResult
+                               maxResults:(NSInteger)maxResults {
+  XCTAssertTrue([detectionResult.detections count] <= maxResults);
+  [self verfiyDetection:detectionResult.detections[0]
+      expectedBoundingBox:CGRectMake(54, 396, 393, 199)
+       expectedFirstScore:0.632812
+       expectedFirstLabel:@"cat"];
+  [self verfiyDetection:detectionResult.detections[1]
+      expectedBoundingBox:CGRectMake(602, 157, 394, 447)
+       expectedFirstScore:0.609375
+       expectedFirstLabel:@"cat"];
+  [self verfiyDetection:detectionResult.detections[2]
+      expectedBoundingBox:CGRectMake(260, 394, 179, 209)
+       expectedFirstScore:0.5625
+       expectedFirstLabel:@"cat"];
+}
+
+- (void)testSuccessfullObjectDetectionOnMLImageWithUIImage {
+  TFLObjectDetectorOptions *objectDetectorOptions =
+      [[TFLObjectDetectorOptions alloc] initWithModelPath:self.modelPath];
+
+  TFLObjectDetector *objectDetector =
+      [TFLObjectDetector objectDetectorWithOptions:objectDetectorOptions error:nil];
+  XCTAssertNotNil(objectDetector);
+
+  GMLImage *gmlImage = [GMLImage imageFromBundleWithClass:[self class]
+                                                 fileName:@"cats_and_dogs"
+                                                   ofType:@"jpg"];
+  XCTAssertNotNil(gmlImage);
+
+  TFLDetectionResult *detectionResults = [objectDetector detectWithGMLImage:gmlImage error:nil];
+  [self verifyResults:detectionResults];
+}
+
+- (void)testModelOptionsWithMaxResults {
+  TFLObjectDetectorOptions *objectDetectorOptions =
+      [[TFLObjectDetectorOptions alloc] initWithModelPath:self.modelPath];
+  int maxResults = 3;
+  objectDetectorOptions.classificationOptions.maxResults = maxResults;
+
+  TFLObjectDetector *objectDetector =
+      [TFLObjectDetector objectDetectorWithOptions:objectDetectorOptions error:nil];
+  XCTAssertNotNil(objectDetector);
+
+  GMLImage *gmlImage = [GMLImage imageFromBundleWithClass:[self class]
+                                                 fileName:@"cats_and_dogs"
+                                                   ofType:@"jpg"];
+  XCTAssertNotNil(gmlImage);
+
+  TFLDetectionResult *detectionResults = [objectDetector detectWithGMLImage:gmlImage error:nil];
+  [self verifyResultsWithMaxResultsOption:detectionResults maxResults:maxResults];
+}
+
+@end
```

Comment:
@lu-wang-g Should we include tests for options in TFLClassificationOptions here (e.g. threshold, allowList, denyList etc.) or leave it as-is? 

I think the ObjC -> C conversion logic of TFLClassificationOptions is already covered in the image classification tests, and the implementation correctness of the underlying C and C++ implementation is covered in tests of each layer, so I think it's fine to keep it as what Prianka has implemented. 

WDYT?